src/Microsoft.Health.Fhir.Core/Features/Search/CodeSystemResolver.cs

```diff
@@ -44,11 +46,18 @@ public string ResolveSystem(string shortPath)
             return null;
         }
 
-        public void Start()
+        public Task StartAsync(CancellationToken cancellationToken)
         {
             using Stream file = _modelInfoProvider.OpenVersionedFileStream("resourcepath-codesystem-mappings.json");
             using var reader = new StreamReader(file);
             _dictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadToEnd());
+
+            return Task.CompletedTask;
+        }
+
+        Task IHostedService.StopAsync(CancellationToken cancellationToken)
```

Comment:
Some thought for future, it might have been nice to have an abstract base so the empty `StopAsync` this is not needed everywhere